### PR TITLE
Support updating a common property that specifies package versions

### PR DESCRIPTION
**Pull request by jasonmalinowski**

### CODE REVIEW

File: src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs [hunk 1]
```diff
@@ -135,14 +142,32 @@ public List<GitChange> SaveComponents()
...
                 .Where(p => p.Attribute("Include")?.Value == packageInfo.PackageName || p.Attribute("Update")?.Value == packageInfo.PackageName)
                 .Select(x => x.Attribute("Version")).SingleOrDefault();
+
+            if (attribute is null)
+                return null;
+
+            // If the attribute is a property substitution, then we'll see if we actually can find the property itself
+            if (attribute.Value.StartsWith("$(") && attribute.Value.EndsWith(")"))
+            {
+                var propertyName = attribute.Value.Substring(2, attribute.Value.Length - 3);
```
Comment (dibarbet) [inline src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs:163]: are we able to use ranges in this project?

File: src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs [hunk 2]
```diff
@@ -183,8 +208,11 @@ public void UpdatePackageVersion(PackageInfo packageInfo)
                     (string? original, XDocument? document) pair = (null, null);
                     if (PackagePropFileToDocumentMap.TryGetValue(file, out pair!))
                     {
-                        var propsVersionAttribute = GetVersionAttributeInPropsFileOpt(pair.document!, packageInfo);
-                        propsVersionAttribute?.SetValue(packageInfo.Version.ToString());
+                        var versionObject = GetVersionAttributeOrElementInPropsFile(pair.document!, packageInfo);
+                        if (versionObject is XAttribute attribute)
+                            attribute.SetValue(packageInfo.Version.ToString());
+                        else if (versionObject is XElement element)
+                            element.SetValue(packageInfo.Version.ToString());
```
Comment (dibarbet) [inline src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs:215]: should we do something (throw / log) if neither is true?

**Review by dibarbet — APPROVED** [review]

File: src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs:163]: We're targeting net46 so no idea. I'd also have to learn how to use ranges for this. :smile:

File: src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs (hunk 2 above)
Comment (jasonmalinowski) [reply src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs:215]: Ah, probably. I don't understand why the null ?. was there before, but you're right this makes it worse if the function were to return non-null but some other type.

**Review by arkalyanms — APPROVED** [review]

File: src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs (hunk 2 above)
Comment (jasonmalinowski) [reply src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs:215]: Added a throw.

**Pull request merged**